File: src/main/java/com/uet/productionmove/entity/ProductStatus.java
package com.uet.productionmove.entity;

/**
 * Trạng thái của sản phẩm
 */
public class ProductStatus {
    //  Vừa mới được sản xuất
    public static String NEWLY_PRODUCED = "Newly produced";
    // Đã đưa về đại lý
    public static String AGENCY = "Agency";
    //  Đã bán cho khách hàng
    public static String SOLD = "Sold";
    //  Lỗi, cần  bảo hành
    public static String ERROR_WARRANTY = "Error need warranty";
    //  Đang được bảo hành
    public static String UNDER_WARRANTY = "Under warranty";
    //  Đã bảo hành xong
    public static String DONE_WARRANTY = "Done warranty";
    //  Đã trả lại khách hàng sau khi bảo hành
    public static String RETURNED_WARRANTY = "Returned warranty";
    // Lỗi, cần trả về nhà máy
    public static String ERROR_FACTORY = "Error factory";
    //  Lỗi, đã đưa về cơ sở sản xuất
    public static String ERROR_RETURNED_FACTORY = "Error returned factory";
    //  Lỗi, cần triệu hồi
    public static String ERROR_SUMMON = "Error summon";
    //  Hết thời gian bảo hành
    public static String WARRANTY_EXPIRED = "Warranty expired";
    //  Trả lại cơ sở sản xuất
    public static String RETURNED_FACTORY = "Returned factory";
}

File: src/main/reactjs/src/pages/Data/ProductStatus.js
const ProductStatus = [
    { key: "Agency", value: "Đang bán"},
    { key: "Sole", value: "Đã bán"},
    { key: "Error need warranty", value: "Lỗi cần bảo hành"},
    { key: "Under warranty", value: "Đang bảo hành"},
    { key: "Returned warranty", value: "Đã trả lại khách hàng"},
    { key: "Error factory", value: "Lỗi cần đưa về nhà máy"},
    { key: "Error returned factory", value: "Lỗi, đã đưa về nhà máy"},
    { key: "Error summon", value: "Lỗi, cần triệu hồi"},
    { key: "Warranty expired", value: "Hết bảo hành"},
    { key: "Returned factory", value: "Đã trả lại nhà máy"},
    { key: "Cannot sale", value: "Không bán được"},
    { key: "Newly produced", value: "Mới sản xuất"},
    { key: "Admin", value: "Ban quản lý"},
    { key: "Distributor", value: "Đại lý"},
    { key: "Warranty", value: "Trung tâm bảo hành"}
];

ProductStatus["Newly produced"] = "Mới sản xuất";
ProductStatus["Agency"] = "Đang bán"
ProductStatus["Sole"] = "Đã bán"
ProductStatus["Error need warranty"] = "Lỗi cần bảo hành"
ProductStatus["Under warranty"] = "Đang bảo hành"
ProductStatus["Returned warranty"] = "Đã trả lại khách hàng"
ProductStatus["Error factory"] = "Lỗi cần đưa về nhà máy"
ProductStatus["Error returned factory"] = "Lỗi, đã đưa về nhà máy"
ProductStatus["Error summon"] = "Lỗi, cần triệu hồi"
ProductStatus["Warranty expired"] = "Hết bảo hành"
ProductStatus["Returned factory"] = "Đã trả lại nhà máy"
ProductStatus["Cannot sale"] = "Không bán được"
ProductStatus["Newly produced"] = "Mới sản xuất"
ProductStatus["Admin"] = "Ban quản lý"
ProductStatus["Distributor"] = "Đại lý"
ProductStatus["Warranty"] = "Trung tâm bảo hành"

export default ProductStatus;